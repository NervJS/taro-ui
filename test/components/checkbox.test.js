import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtCheckbox from '../../.temp/components/checkbox/index'

const checkboxOption = [
  {
    value: 'list1',
    label: 'iPhone X',
    desc:
      '部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。'
  },
  { value: 'list2', label: 'HUAWEI P20' },
  {
    value: 'list3',
    label: 'OPPO Find X',
    desc:
      '部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。',
    disabled: true
  },
  {
    value: 'list4',
    label: 'vivo NEX',
    desc:
      '部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。',
    disabled: true
  }
]

describe('AtCheckbox Snap', () => {
  it('render initial AtCheckbox', () => {
    const component = renderToString(<AtCheckbox />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCheckbox -- props options', () => {
    const component = renderToString(<AtCheckbox options={checkboxOption} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtCheckbox -- props selectedList', () => {
    const component = renderToString(
      <AtCheckbox options={checkboxOption} selectedList={['list2']} />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('AtCheckbox Event', () => {
  it('AtCheckbox onChange', () => {
    const onClick = jest.fn()
    const component = renderIntoDocument(
      <AtCheckbox
        options={checkboxOption}
        selectedList={['list2']}
        onChange={onClick}
      />
    )
    const componentDom = findDOMNode(component, 'at-checkbox')
    const items = componentDom.querySelectorAll('.at-checkbox__option')
    Simulate.click(items[0])
    expect(onClick).toBeCalled()
    expect(onClick.mock.calls[0][0]).toEqual(['list2', 'list1'])
  })

  it('AtCheckbox disabled, onChange not to be called', () => {
    const onClick = jest.fn()
    const component = renderIntoDocument(
      <AtCheckbox
        options={checkboxOption}
        selectedList={['list2']}
        onChange={onClick}
      />
    )
    const componentDom = findDOMNode(component, 'at-checkbox')
    const items = componentDom.querySelectorAll('.at-checkbox__option')
    Simulate.click(items[2])
    expect(onClick).not.toBeCalled()
  })
})
