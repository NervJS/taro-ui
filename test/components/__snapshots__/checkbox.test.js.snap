// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtCheckbox Snap render AtCheckbox -- props options 1`] = `"<div class=\\"at-checkbox\\"><div class=\\"at-checkbox__option\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">iPhone X</div></div><div class=\\"at-checkbox__desc\\">部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。</div></div></div><div class=\\"at-checkbox__option\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">HUAWEI P20</div></div>undefined</div></div><div class=\\"at-checkbox__option at-checkbox__option--disabled\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">OPPO Find X</div></div><div class=\\"at-checkbox__desc\\">部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。</div></div></div><div class=\\"at-checkbox__option at-checkbox__option--disabled\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">vivo NEX</div></div><div class=\\"at-checkbox__desc\\">部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。</div></div></div></div>"`;

exports[`AtCheckbox Snap render AtCheckbox -- props selectedList 1`] = `"<div class=\\"at-checkbox\\"><div class=\\"at-checkbox__option\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">iPhone X</div></div><div class=\\"at-checkbox__desc\\">部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。</div></div></div><div class=\\"at-checkbox__option at-checkbox__option--selected\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">HUAWEI P20</div></div>undefined</div></div><div class=\\"at-checkbox__option at-checkbox__option--disabled\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">OPPO Find X</div></div><div class=\\"at-checkbox__desc\\">部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。</div></div></div><div class=\\"at-checkbox__option at-checkbox__option--disabled\\"><div class=\\"at-checkbox__option-wrap\\"><div class=\\"at-checkbox__option-cnt\\"><div class=\\"at-checkbox__icon-cnt\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div><div class=\\"at-checkbox__title\\">vivo NEX</div></div><div class=\\"at-checkbox__desc\\">部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。</div></div></div></div>"`;

exports[`AtCheckbox Snap render initial AtCheckbox 1`] = `"<div class=\\"at-checkbox\\"></div>"`;
